diff --git a/libutil/tty.c b/libutil/tty.c
index bceb01e749..8a2cd73e78 100644
--- a/libutil/tty.c
+++ b/libutil/tty.c
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #include <fcntl.h>
 #include <dirent.h>
diff --git a/mknod.c b/mknod.c
index 8de35c712c..27f91e0c77 100644
--- a/mknod.c
+++ b/mknod.c
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 
 #include <fcntl.h>
 #include <stdio.h>
diff --git a/mountpoint.c b/mountpoint.c
index 8f205a271f..fd3a13300c 100644
--- a/mountpoint.c
+++ b/mountpoint.c
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 
 #include <mntent.h>
 #include <stdio.h>
diff --git a/stat.c b/stat.c
index 220a659602..40b372dd36 100644
--- a/stat.c
+++ b/stat.c
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 
 #include <inttypes.h>
 #include <stdio.h>
